thunder/benchmarks/__init__.py

```diff
@@ -2958,3 +2960,185 @@ def fn(self) -> Callable:
 #     if args.listbenchmarks:
 #         list_benchmarks(use_classname=False)
 #         sys.exit(0)
+
+from thunder.dynamo.utils import _concrete_shape
+from thunder.dynamo.compiler import ThunderCompiler
+from itertools import chain
+
+GRAPH_BY_GRAPH_BENCHMARK_PARAMS_KEYS = ("GraphID", "SplitModuleName", "executor")
+
+
+def _get_example_inputs_from_placeholder(node) -> tuple[torch.Tensor]:
+    check(node.op == "placeholder", lambda: f"The node must be placeholder type", ValueError)
+    check("example_value" in node.meta, lambda: "example_value does not exist in the meta of {node}", ValueError)
+    example_value = node.meta["example_value"]
+
+    if isinstance(example_value, torch.Tensor):
+        return (
+            make_tensor(
+                _concrete_shape(example_value),
+                dtype=example_value.dtype,
+                device=example_value.device,
+                requires_grad=example_value.requires_grad,
+            ),
+        )
+    elif isinstance(example_value, tuple):
+        return tuple(
+            make_tensor(
+                _concrete_shape(e_v),
+                dtype=e_v.dtype,
+                device=e_v.device,
+                requires_grad=e_v.requires_grad,
+            )
+            for e_v in example_value
+        )
+    else:
+        raise TypeError(
+            "The 'example_value' in the placeholder node is expected to be either a Tensor or a Tuple of Tensors."
+        )
+
+
+class ThunderCompilerGraphBenchmarking(ThunderCompiler):
+    _executors = (
+        "eager",
+        "inductor",
+        "thunder",
+    )
+
+    def __init__(
+        self,
+        bench: BenchmarkFixture,
+        executors: Sequence[str] = None,
```

Comment:
can't we just 
```suggestion
        executors: Sequence[str],
```
?

Reply:
yes, I can rm it, I just thought there are https://github.com/Lightning-AI/lightning-thunder/blob/b6a8940cda2ba564eac3b1df6f2201dd954750f5/thunder/benchmarks/__init__.py#L3078-L3079, so it's not much harm to keep the None here